game/shared/ehandle.h

```diff
@@ -11,31 +11,19 @@
 #pragma once
 #endif
 
-#if defined( _DEBUG ) && defined( GAME_DLL )
-#include "tier0/dbg.h"
-#include "cbase.h"
-#endif
-
-
-#include "const.h"
-#include "basehandle.h"
-#include "entitylist_base.h"
-
-
-class IHandleEntity;
-
+#include "entityhandle.h"
+#include "entitysystem.h"
 
 // -------------------------------------------------------------------------------------------------- //
 // Game-code CBaseHandle implementation.
 // -------------------------------------------------------------------------------------------------- //
 
-inline IHandleEntity* CBaseHandle::Get() const
+inline IHandleEntity* CEntityHandle::GetHandleEntity() const
 {
-	extern CBaseEntityList *g_pEntityList;
-	return g_pEntityList->LookupEntity( *this );
+	extern CEntitySystem *g_pEntitySystem;
+	return g_pEntitySystem->GetBaseEntity( *this );
 }
 
-
 // -------------------------------------------------------------------------------------------------- //
 // CHandle.
 // -------------------------------------------------------------------------------------------------- //

```

Comment:
CHandle should inherit from CEntityHandle as well as refactored for the useage.

Reply:
CHandle inherits from CBaseHandle which is typedef'd to CEntityHandle, but I changed it anyway just to be clearer.

Reply:
dwarf bin has it defined like that, so I'd stick to that as well